django/db/models/constraints.py

```diff
@@ -1,17 +1,28 @@
 from enum import Enum
 
-from django.db.models.expressions import ExpressionList, F
+from django.db import connections
+from django.core.exceptions import FieldError, ValidationError
+from django.db.models.expressions import (
+    Exists, ExpressionList, F, Value,
+)
 from django.db.models.indexes import IndexExpression
+from django.db.models.lookups import Exact
 from django.db.models.query_utils import Q
 from django.db.models.sql.query import Query
+from django.db.utils import DEFAULT_DB_ALIAS
+from django.utils.translation import gettext_lazy as _
 from django.utils.version import PY310
 
 __all__ = ['CheckConstraint', 'Deferrable', 'UniqueConstraint']
 
 
 class BaseConstraint:
-    def __init__(self, name):
+    message = _('A constraint failed.')
+
+    def __init__(self, name, message=None):
```

Comment:
Wonder if we should have a more explicit name here, `message` makes sense in the case of `Validator` but it seems a bit ambiguous in the case of constraint. Thoughts @django/triage-review?

Reply:
As far as I can see `message` is always `None`. It doesn't seem to be used. Can it just be removed?

Reply:
The point of it is for the `ValidationErrror` raised to use this message just like `Validator(message)` is used

https://github.com/django/django/blob/910ecd1b8df7678f45c3d507dde6bcb1faafa243/django/core/validators.py#L27-L51

Reply:
We should add a test for passing a `message` to the constraint.

Reply:
I added tests on `BaseConstraint.get_violation_error_message`. Should we also test passing custom messages for the implemented constraints (`CheckConstraint`, ...) as they only pass it through `super`?